velox/vector/VariantToVector.cpp

```diff
@@ -85,12 +118,24 @@ ArrayVectorPtr variantArrayToVector(
     }
   }
   if (variantArray.empty() || !nonNullIndex.has_value()) {
-    return variantArrayToVectorImpl<TypeKind::UNKNOWN>(variantArray, pool);
+    return variantArrayToVectorImpl<TypeKind::UNKNOWN>(
+        variantArray, pool, std::nullopt);
   }
 
   const auto elementType = variantArray.at(nonNullIndex.value()).inferType();
+  if (elementType->kind() == TypeKind::SHORT_DECIMAL) {
+    return variantArrayToVectorImpl<TypeKind::SHORT_DECIMAL>(
+        variantArray, pool, type);
+  } else if (elementType->kind() == TypeKind::LONG_DECIMAL) {
+    return variantArrayToVectorImpl<TypeKind::LONG_DECIMAL>(
+        variantArray, pool, type);
+  }
   return VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
```

Comment:
`#define VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH_ALL(TEMPLATE_FUNC, typeKind, ...)`

Use this macro. It includes decimals.

Reply:
Thanks for the suggestion, I am running into some issues using this macro and am checking further.

Reply:
Because unscaled decimal value is needed to construct the arrayElements for decimals, using ```VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH_ALL``` results in a compilation error at
```T(variantArray[i].value<KIND>())```.